packages/mask/src/social-network-adaptor/mirror.xyz/collecting/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { ProfileIdentifier } from '@masknet/shared-base'
+import type { Writer } from '@masknet/web3-providers'
+import urlcat from 'urlcat'
+import { mirrorBase } from '../base.js'
+
+export const getMirrorProfileUrl = (id: string) => urlcat('https://mirror.xyz/:id', { id })
+
+export function formatWriter(writer: Writer) {
+    return {
+        avatar: writer.avatarURL,
+        nickname: writer.displayName,
+        bio: writer.description,
+        homepage: writer.domain || getMirrorProfileUrl(writer.address),
+        identifier: ProfileIdentifier.of(mirrorBase.networkIdentifier, writer.address).unwrapOr(undefined),
+    }
+}
+
+export enum MirrorPageType {
+    Profile = 'profile',
+    Collection = 'collection',
+    Post = 'post',
+    Dashboard = 'dashboard',
+}
+
+const ADDRESS_FULL = /0x\w{40,}/i
+const MIRROR_SUB_DOMAIN = /https(.*)mirror.xyz/i
```

Comment:
Did you forget that?

```suggestion
const MIRROR_SUB_DOMAIN = /https:\/\/(.*)mirror\.xyz/i
```

Reply:
It's even better if we can elaborate the `.*` into more specific chars.